client/src/charts/nivo/NivoCharts.js

```diff
@@ -0,0 +1,1037 @@
+            };
+
+            if (include_percent){
+              return percent_value_tooltip(
+                [data_with_original_values],
+                get_formatter(is_money, text_formatter, false), 
+                total
+              );
+            } else {
+              return tooltip(
+                [data_with_original_values],
+                get_formatter(is_money, text_formatter, false)
+              );
+            } 
+          }}
+          innerRadius={0.5}
+          borderWidth={1}
+          borderColor="inherit:darker(0.2)"
+          radialLabelsSkipAngle={0}
+          animate={true}
+          motionStiffness={30}
+          motionDamping={15}
+        />
+      </div>
+      <div className="common-donut__legend">
+        <div className="centerer">
+          <div className="centerer-IE-fix">
+            <TabularPercentLegend
+              items={legend_items}
+              get_right_content={
+                (item) => (
+                  <div>
+                    <span className="common-donut__legend-data">
+                      <Format type="compact1" content={item.value} />
+                    </span>
+                    <span className="common-donut__legend-data">
+                      <Format type="percentage1" content={(item.value)*Math.pow(legend_total,-1)} />
+                    </span>
+                  </div>
+                )
+              }
+            />
+          </div>
+        </div>
+      </div>
+    </div>;
+    
+
+    return <InteractiveGraph graph={graph} table={table} />;
+  }
+}
+NivoResponsivePie.defaultProps = {
+  ...general_default_props,
+  margin: {
+    top: 30,
+    right: 80,
+    bottom: 60,
+    left: 50,
+  },
+  include_percent: true,
+  enableRadialLabels: false,
+  enableSlicesLabels: false,
+};
+
+export class NivoResponsiveBar extends React.Component{
+  render(){
+    const{
+      data,
+      keys,
+      margin,
+      label_format,
+      colors,
+      bttm_axis,
+      left_axis,
+      isInteractive,
+      indexBy,
+      remove_bottom_axis,
+      remove_left_axis,
+      enableLabel,
+      label,
+      is_money,
+      legends,
+      tick_value,
+      text_formatter,
+      theme,
+      colorBy,
+      min,
+      max,
+      motion_damping,
+      motion_stiffness,
+      tooltip,
+      enableGridX,
+      groupMode,
+      enableGridY,
+      onMouseEnter,
+      onMouseLeave,
+      onClick,
+      padding,
+      animate,
+      labelTextColor,
+      borderWidth,
+      disable_table_view,
+    } = this.props;
+
+    legends && (legends[0].symbolShape = fixedSymbolShape);
+
+    const table_data = bar_table_data(data, indexBy, get_formatter(is_money, text_formatter, true, true));
+    const table_header_keys = _.concat([indexBy],keys);
+
+    const table = !disable_table_view && <DisplayTable data={table_data} column_keys={table_header_keys} sort_keys={table_header_keys} table_data_headers={table_header_keys} table_name={"TODO"}/>;
```

Comment:
I think you should break this up! Same for all below

Reply:
Which line? As I said above line 431 is going to change because the DisplayTable API changed.

Reply:
ohhh okay didn't see that